libraries/rush-lib/src/cli/RushXCommandLine.ts

```diff
@@ -45,129 +47,139 @@ interface IRushXCommandLineArguments {
 }
 
 export class RushXCommandLine {
-  public static launchRushX(launcherVersion: string, isManaged: boolean): void {
-    RushXCommandLine._launchRushXInternal(launcherVersion, { isManaged });
+  public static launchRushX(launcherVersion: string, options: ILaunchRushXInternalOptions): void {
+    try {
+      const args: IRushXCommandLineArguments = this._getCommandLineArguments();
+      const rushConfiguration: RushConfiguration | undefined = RushConfiguration.tryLoadFromDefaultLocation({
+        showVerbose: false
+      });
+      const eventHooksManager: EventHooksManager | undefined = rushConfiguration
+        ? new EventHooksManager(rushConfiguration)
+        : undefined;
+
+      const ignoreHooks = process.env.INSIDERUSH === '1';
+      eventHooksManager?.handle(Event.preRushx, false /* isDebug */, ignoreHooks);
+      RushXCommandLine._launchRushXInternal(launcherVersion, options, rushConfiguration, args);
+      eventHooksManager?.handle(Event.postRushx, false /* isDebug */, ignoreHooks);
+    } catch (error) {
+      console.log(colors.red('Error: ' + (error as Error).message));
+    }
   }
 
   /**
    * @internal
    */
-  public static _launchRushXInternal(launcherVersion: string, options: ILaunchRushXInternalOptions): void {
+  public static _launchRushXInternal(
+    launcherVersion: string,
+    options: ILaunchRushXInternalOptions,
+    rushConfiguration: RushConfiguration | undefined,
+    args: IRushXCommandLineArguments
```

Comment:
In our convention, usually `options` is the last parameter, and generally it has most or all of the parameters.

Would it make sense to move all these parameters into `ILaunchRushXInternalOptions`?

Reply:
fixed